user_manual.md

```diff
@@ -1,146 +1,177 @@
 Further output can be given by user-provided `echo` commands.
 
 ### Streaming input to the Ethos
 
 The `ethos` binary accepts input piped from stdin. The following are all equivalent ways of running `ethos`:
-```
+
+```shell
 % ethos <file>
 % ethos < <file>
 % cat <file> | ethos
 ```
 
-## Overview of Features
+## Overview of Eunoia's features
+
+Eunoia is the name of the logical framework and language that is supported natively by the Ethos checker.
+Although it is a general logical framework, Eunoia is geared towards defining proof systems used by SMT solvers and writing proofs in those systems.
 
-Eunoia is the name of the logical framework that is the native language accepted by the Ethos checker.
-Eunoia is a logical framework for defining proof rules and writing proofs from SMT solvers. Eunoia files typically are given the suffix `*.eo`.
+Eunoia files are text file that are typically given the suffix `*.eo`.
 
 The core features of Eunoia include:
-- Support for SMT-LIB version 3.0 syntax for defining theory signatures.
-- Support for associating syntactic categories (as specified by SMT-LIB e.g. `<numeral>`) with types.
-- A command `declare-rule` for defining proof rules.
-- A set of commands for specifying proofs (`step`, `assume`, and so on), whose syntax closely follows the Alethe proof format (for details see [here](https://verit.gitlabpages.uliege.be/alethe/specification.pdf)).
-- A library of operations (`eo::add`, `eo::mul`, `eo::concat`, `eo::extract`) for performing computations over values.
-- A command `program` for defining side conditions as ordered list of rewrite rules.
-- A command `declare-oracle-fun` for user-provided oracles, that is, functions whose semantics are given by external binaries. Oracles can be used e.g. for modular proof checking.
-- Commands for file inclusion (`include`) and referencing (`reference`). The latter command can be used to specify the name of an `*.smt2` input file that the proof is associated with.
 
-In the following sections, we review these features in more detail. A full syntax for the commands is given at the end of this document.
+- A syntax and type system for defining theory signatures that follows those of SMT-LIB version 3.0.
+- Constructs for associating syntactic categories (as specified by SMT-LIB e.g. `<numeral>`) with types.
+- A command, `declare-rule`, for defining proof rules.
+- A set of commands for specifying proofs (`step`, `assume`, and so on), whose syntax closely follows that of the Alethe proof format (for details, see [here](https://verit.gitlabpages.uliege.be/alethe/specification.pdf)).
+- A set of built-in basic types and a library of operations (`eo::add`, `eo::mul`, `eo::concat`, `eo::extract`) for performing computations over values.
+  <!--CT It would be more consistent with the rest of the terminology to use `define-program` instead -->
```

Comment:
Agreed, we can maybe rename this in a future stable release.

Reply:
Sounds good.